lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+			propertyKeyName: zwaveValue.propertyKeyName,
+			type: zwaveValueMeta.type, // https://github.com/zwave-js/node-zwave-js/blob/cb35157da5e95f970447a67cbb2792e364b9d1e1/packages/core/src/values/Metadata.ts#L28
+			readable: zwaveValueMeta.readable,
+			writeable: zwaveValueMeta.writeable,
+			description: zwaveValueMeta.description,
+			label:
+				zwaveValueMeta.label || zwaveValue.propertyName + ' (property)', // when label is missing, re use propertyName. Usefull for webinterface
+			default: zwaveValueMeta.default,
+			ccSpecific: zwaveValueMeta.ccSpecific,
+			stateless: zwaveValue.stateless || false, // used for notifications to specify that this should not be persisted (retained)
+		}
+
+		// Value types: https://github.com/zwave-js/node-zwave-js/blob/cb35157da5e95f970447a67cbb2792e364b9d1e1/packages/core/src/values/Metadata.ts#L28
+		if (zwaveValueMeta.type === 'number') {
+			valueId.min = (zwaveValueMeta as ValueMetadataNumeric).min
+			valueId.max = (zwaveValueMeta as ValueMetadataNumeric).max
+			valueId.step = (zwaveValueMeta as ValueMetadataNumeric).steps
+			valueId.unit = (zwaveValueMeta as ValueMetadataNumeric).unit
+		} else if (zwaveValueMeta.type === 'string') {
+			valueId.minLength = (
+				zwaveValueMeta as ValueMetadataString
+			).minLength
+			valueId.maxLength = (
+				zwaveValueMeta as ValueMetadataString
+			).maxLength
+		}
+
+		if (utils.hasProperty(zwaveValueMeta, 'states')) {
+			valueId.list = true
+			valueId.allowManualEntry = (
+				zwaveValueMeta as ConfigurationMetadata
+			).allowManualEntry
+			valueId.states = []
+			for (const k in (zwaveValueMeta as ValueMetadataNumeric).states) {
+				valueId.states.push({
+					text: (zwaveValueMeta as ValueMetadataNumeric).states[k],
+					value: parseInt(k),
+				})
+			}
+		} else {
+			valueId.list = false
+		}
+
+		return valueId
+	}
+
+	/**
+	 * Add a node value to our node values
+	 *
+	 */
+	_addValue(
+		zwaveNode: ZWaveNode,
+		zwaveValue: TranslatedValueID,
+		oldValues?: {
+			[key: string]: Z2MValueId
+		}
+	) {
+		const node = this.nodes.get(zwaveNode.id)
+
+		if (!node) {
+			logger.info(`ValueAdded: no such node: ${zwaveNode.id} error`)
+		} else {
+			const zwaveValueMeta = zwaveNode.getValueMetadata(zwaveValue)
+
+			const valueId = this._parseValue(
+				zwaveNode,
+				zwaveValue,
+				zwaveValueMeta
+			)
+
+			const vID = this._getValueID(valueId)
+
+			// a valueId is udpated when it doesn't exist or its value is updated
+			const updated =
+				!oldValues ||
+				!oldValues[vID] ||
+				oldValues[vID].value !== valueId.value
+
+			logger.info(
+				`Node ${zwaveNode.id}: value added ${valueId.id} => ${valueId.value}`
+			)
+
+			if (updated) {
+				this.emit('valueChanged', valueId, node)
+			}
+		}
+	}
+
+	/**
+	 * Parse a zwave value into a valueID
+	 *
+	 * @param { ZWaveNode } zwaveNode
+	 * @param { TranslatedValueID } zwaveValue
+	 */
+	_parseValue(
+		zwaveNode: ZWaveNode,
+		zwaveValue: TranslatedValueID & { [x: string]: any },
```

Comment:
What's this weird type change for?

Reply:
There could be some additional properties in zwaveValue